internal/filter/filter.go

```diff
@@ -177,12 +184,8 @@ func List(patterns []string, str string) (matched bool, childMayMatch bool, err
 			return false, false, err
 		}
 
-		matched = matched || m
-		childMayMatch = childMayMatch || c
-
-		if matched && childMayMatch {
-			return true, true, nil
-		}
+		matched = matched && !m || m && !negate
+		childMayMatch = childMayMatch && !c || c && !negate
```

Comment:
I believe this should be:
```suggestion
		childMayMatch = childMayMatch || c && !negate
```

I believe the issue is that the `childMayMatch` should not reset in the same way as the primary match term - it should retain the behaviour of triggering to true if *any* pattern indicates that a child may match.